server/parameters/util/YAMLParser.java

```diff
@@ -420,21 +439,124 @@ private static long parse(String durationString) {
                     String lenStr = matcher.group(LENGTH_GROUP);
                     String unitStr = matcher.group(UNIT_GROUP);
                     long lenValue = Long.parseLong(lenStr);
-                    if (unitStr.equalsIgnoreCase("d")) duration = java.time.Duration.of(lenValue, ChronoUnit.DAYS);
-                    else if (unitStr.equalsIgnoreCase("h")) duration = java.time.Duration.of(lenValue, ChronoUnit.HOURS);
-                    else if (unitStr.equalsIgnoreCase("m")) duration = java.time.Duration.of(lenValue, ChronoUnit.MINUTES);
-                    else if (unitStr.equalsIgnoreCase("s")) duration = java.time.Duration.of(lenValue, ChronoUnit.SECONDS);
-                    else throw new IllegalStateException("Unexpected duration unit: " + unitStr);
+                    if (unitStr.equalsIgnoreCase("d")) {
+                        duration = java.time.Duration.of(lenValue, ChronoUnit.DAYS);
+                    } else if (unitStr.equalsIgnoreCase("h")) {
+                        duration = java.time.Duration.of(lenValue, ChronoUnit.HOURS);
+                    } else if (unitStr.equalsIgnoreCase("m")) {
+                        duration = java.time.Duration.of(lenValue, ChronoUnit.MINUTES);
+                    } else if (unitStr.equalsIgnoreCase("s")) {
+                        duration = java.time.Duration.of(lenValue, ChronoUnit.SECONDS);
+                    } else throw new IllegalStateException("Unexpected duration unit: " + unitStr);
                     return duration.toSeconds();
                 } else {
                     throw new IllegalArgumentException("Duration [" + durationString + "] is not in a recognised format.");
                 }
             }
 
-            private static boolean isValidDurationString(String durationString) {
+            private static boolean isValidString(String durationString) {
                 Matcher matcher = DURATION_PATTERN.matcher(durationString);
                 return matcher.matches();
             }
         }
+
+        public enum TimePeriodName {
```

Comment:
New enums and parameters for the config. It's not very clean to depend on YAML parser values directly from outside, we should move them elsewhere